docusaurus.config.js

```diff
@@ -63,12 +63,24 @@ const config = {
   themeConfig:
     /** @type {import('@docusaurus/preset-classic').ThemeConfig} */
     ({
-      algolia: {
-        appId: process.env.ALGOLIA_APP_ID,
-        apiKey: process.env.ALGOLIA_SEARCH_API_KEY,
-        indexName: process.env.ALGOLIA_INDEX_NAME,
-        searchPagePath: false,
-      },
+
+	// Checks if all environment variables are in place. Only adds to config
+	    // if they it all passes
+	
+	
+      	...(process.env.ALGOLIA_APP_ID && 
```

Comment:
I would do this a little differently. I suggest creating an object above:
```
const ALGOLIA_SECRETS = {
  appId: process.env.ALGOLIA_APP_ID,
  apiKey: process.env.ALGOLIA_SEARCH_API_KEY,
  indexName: process.env.ALGOLIA_INDEX_NAME,
}
```
I would then create a function:
```
function getAlgoliaConfig(config) {
  if (ALGOLIA_SECRETS.some(secret => secret === undefined)) {
    return undefined;
  }

  return {...ALGOLIA_SECRETS, ...config};
}
```
And then would use this in the `themeConfig` as so:
```
themeConfig: {
  algolia: getAlgoliaConfig({ searchPagePath: false }),
  ...
}
```
This will make it easier to extend, remove or change `algolia` secrets if needed and declutter the general `docusaurus` configuration.

Also, next time I would suggest opening separate PRs for separate issues.

Don't forget to run `npm run lint` and `npm run format` to lint and prettify the code.